charts/reporting-operator/templates/reporting-operator-deployment.yaml

```diff
@@ -156,6 +156,10 @@ spec:
               name: reporting-operator-config
               key: prometheus-datasource-import-from
               optional: true
+{{- /* neither specified = no auth used; both specified = error; either = correct & authenticated */ -}}
+{{- if and .Values.spec.config.prometheusImporter.auth.tokenSecret.enabled .Values.spec.config.prometheusImporter.auth.useServiceAccountToken  }}
+  {{ required "cannot use both token from secret and token from service account" nil }}
```

Comment:
You can use `fail` instead of `required` since you're not using the 2nd argument value